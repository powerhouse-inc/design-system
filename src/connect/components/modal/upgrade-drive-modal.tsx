import { Modal } from '@/powerhouse';
import React, { ComponentPropsWithoutRef } from 'react';
import { Button } from 'react-aria-components';
import { twMerge } from 'tailwind-merge';

const buttonStyles =
    'min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all';

export type ConnectUpgradeDriveModalProps = ComponentPropsWithoutRef<
    typeof Modal
> & {
    header: React.ReactNode;
    body: React.ReactNode;
    onContinue: () => void;
    cancelLabel: string;
    continueLabel: string;
};

export const ConnectUpgradeDriveModal = (
    props: ConnectUpgradeDriveModalProps,
) => {
    const {
        body,
        header,
        onOpenChange,
        onContinue,
        cancelLabel,
        continueLabel,
        overlayProps,
        contentProps,
        ...restProps
    } = props;

    return (
        <Modal
            overlayProps={{
                ...overlayProps,
                className: twMerge('top-10', overlayProps?.className),
            }}
            contentProps={{
                ...contentProps,
                className: twMerge('rounded-3xl', contentProps?.className),
            }}
            onOpenChange={onOpenChange}
            {...restProps}
        >
            <div className="w-[400px] p-6 text-[#565868]">
                <div className="border-b border-[#E5E7E8] pb-2 text-2xl font-bold text-grey-800">
                    {header}
                </div>
                <div className="my-6 rounded-md bg-slate-50 p-4 text-center text-slate-200">
                    {body}
                </div>
                <div className="mt-8 flex justify-between gap-3">
                    <Button
                        onPress={() => onOpenChange?.(false)}
                        className={twMerge(
                            buttonStyles,
                            'flex-1 bg-slate-50 text-[##141718]',
                        )}
                    >
                        {cancelLabel}
                    </Button>
                    <Button
                        onPress={onContinue}
                        className={twMerge(
                            buttonStyles,
                            'flex-1 bg-grey-800 text-[#FEFEFE]',
                        )}
                    >
                        {continueLabel}
                    </Button>
                </div>
            </div>
        </Modal>
    );
};
