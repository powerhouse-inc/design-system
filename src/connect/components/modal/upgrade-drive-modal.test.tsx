import { fireEvent, render, screen } from '@testing-library/react';
import { ConnectUpgradeDriveModal } from './upgrade-drive-modal';

describe('UpgradeDriveModal Component', () => {
    it('should match snapshot', () => {
        render(
            <ConnectUpgradeDriveModal
                open={true}
                onClose={() => {}}
                data-testid="upgrade-drive-modal"
                body="You are moving files from a private to a shared drive. These files will become accessible to others in the shared drive.Do you want to proceed?"
                cancelLabel="Cancel"
                continueLabel="Continue"
                header="Upgrade to cloud drive"
                onContinue={() => {}}
            />,
        );

        const modalComponent = screen.getByTestId('upgrade-drive-modal');

        expect(modalComponent).toMatchSnapshot();
    });

    it('should render modal content', () => {
        const headerText = 'Upgrade to cloud drive';
        const bodyText =
            'ou are moving files from a private to a shared drive. These files will become accessible to others in the shared drive.Do you want to proceed?';
        const cancelLabel = 'Cancel';
        const continueLabel = 'Continue';

        render(
            <ConnectUpgradeDriveModal
                open={true}
                onClose={() => {}}
                data-testid="upgrade-drive-modal"
                body={bodyText}
                cancelLabel={cancelLabel}
                continueLabel={continueLabel}
                header={headerText}
                onContinue={() => {}}
            />,
        );

        const header = screen.getByText(headerText);
        const body = screen.getByText(bodyText);
        const cancelButton = screen.getByText(cancelLabel);
        const deleteButton = screen.getByText(continueLabel);

        expect(header).toBeInTheDocument();
        expect(body).toBeInTheDocument();
        expect(cancelButton).toBeInTheDocument();
        expect(deleteButton).toBeInTheDocument();
    });

    it('should trigger onClose callback when cancel is clicked', () => {
        const onClose = jest.fn();
        const cancelLabel = 'Cancel';

        render(
            <ConnectUpgradeDriveModal
                open={true}
                onClose={onClose}
                data-testid="upgrade-drive-modal"
                cancelLabel={cancelLabel}
                body="You are moving files from a private to a shared drive. These files will become accessible to others in the shared drive.Do you want to proceed?"
                continueLabel="Continue"
                header="Upgrade to cloud drive"
                onContinue={() => {}}
            />,
        );

        fireEvent.click(screen.getByText(cancelLabel));

        expect(onClose).toHaveBeenCalled();
    });

    it('should trigger onDelete callback when CTA is clicked', () => {
        const onDelete = jest.fn();
        const continueLabel = 'Continue';

        render(
            <ConnectUpgradeDriveModal
                open={true}
                onClose={() => {}}
                data-testid="upgrade-drive-modal"
                body="You are moving files from a private to a shared drive. These files will become accessible to others in the shared drive.Do you want to proceed?"
                cancelLabel="Cancel"
                continueLabel={continueLabel}
                header="Upgrade to cloud drive"
                onContinue={onDelete}
            />,
        );

        fireEvent.click(screen.getByText(continueLabel));

        expect(onDelete).toHaveBeenCalled();
    });
});
