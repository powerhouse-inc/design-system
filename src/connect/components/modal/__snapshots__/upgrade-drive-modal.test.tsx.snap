// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UpgradeDriveModal Component > should match snapshot 1`] = `
<div
  aria-describedby="radix-:r2:"
  aria-labelledby="radix-:r1:"
  class="bg-white data-[state=closed]:animate-zoom-out data-[state=open]:animate-zoom-in rounded-3xl"
  data-state="open"
  data-testid="upgrade-drive-modal"
  id="radix-:r0:"
  role="dialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <div
    class="w-[400px] p-6 text-[#565868]"
  >
    <div
      class="border-b border-[#E5E7E8] pb-2 text-2xl font-bold text-[#404446]"
    >
      Upgrade to cloud drive
    </div>
    <div
      class="my-6 rounded-md bg-[#F3F5F7] p-4 text-center text-[#6C7275]"
    >
      You are moving files from a private to a shared drive. These files will become accessible to others in the shared drive.Do you want to proceed?
    </div>
    <div
      class="mt-8 flex justify-between gap-3"
    >
      <button
        class="min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all flex-1 bg-[#F3F5F7] text-[##141718]"
        data-focus-visible="true"
        data-focused="true"
        data-rac=""
        type="button"
      >
        Cancel
      </button>
      <button
        class="min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all flex-1 bg-[#404446] text-[#FEFEFE]"
        data-rac=""
        type="button"
      >
        Continue
      </button>
    </div>
  </div>
</div>
`;
